tsl/src/continuous_aggs/common.c

```diff
@@ -0,0 +1,1583 @@
+static FuncExpr *build_conversion_call(Oid type, FuncExpr *boundary);
+static FuncExpr *build_boundary_call(int32 ht_id, Oid type);
+static Const *cagg_boundary_make_lower_bound(Oid type);
+static Node *build_union_query_quals(int32 ht_id, Oid partcoltype, Oid opno, int varno,
+									 AttrNumber attno);
+static RangeTblEntry *make_subquery_rte(Query *subquery, const char *aliasname);
+
+#define DEFAULT_MATPARTCOLUMN_NAME "time_partition_col"
+
+static inline void
+makeMaterializeColumnName(char *colbuf, const char *type, int original_query_resno, int colno)
+{
+	int ret = snprintf(colbuf, NAMEDATALEN, "%s_%d_%d", type, original_query_resno, colno);
+	if (ret < 0 || ret >= NAMEDATALEN)
+		ereport(ERROR,
+				(errcode(ERRCODE_INTERNAL_ERROR), errmsg("bad materialization table column name")));
+}
+
+/*
+ * Creates a partialize expr for the passed in agg:
+ * partialize_agg(agg).
+ */
+static FuncExpr *
+get_partialize_funcexpr(Aggref *agg)
+{
+	FuncExpr *partialize_fnexpr;
+	Oid partfnoid, partargtype;
+	partargtype = ANYELEMENTOID;
+	partfnoid =
+		LookupFuncName(list_make2(makeString(INTERNAL_SCHEMA_NAME), makeString(TS_PARTIALFN)),
+					   1,
+					   &partargtype,
+					   false);
+	partialize_fnexpr = makeFuncExpr(partfnoid,
+									 BYTEAOID,
+									 list_make1(agg), /*args*/
+									 InvalidOid,
+									 InvalidOid,
+									 COERCE_EXPLICIT_CALL);
+	return partialize_fnexpr;
+}
+
+static Const *
+check_time_bucket_argument(Node *arg, char *position)
+{
+	if (IsA(arg, NamedArgExpr))
+		arg = (Node *) castNode(NamedArgExpr, arg)->arg;
+
+	Node *expr = eval_const_expressions(NULL, arg);
+
+	if (!IsA(expr, Const))
+		ereport(ERROR,
+				(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+				 errmsg("only immutable expressions allowed in time bucket function"),
+				 errhint("Use an immutable expression as %s argument to the time bucket function.",
+						 position)));
+
+	return castNode(Const, expr);
+}
+
+/*
+ * Initialize caggtimebucket.
+ */
+static void
+caggtimebucketinfo_init(CAggTimebucketInfo *src, int32 hypertable_id, Oid hypertable_oid,
+						AttrNumber hypertable_partition_colno, Oid hypertable_partition_coltype,
+						int64 hypertable_partition_col_interval, int32 parent_mat_hypertable_id)
+{
+	src->htid = hypertable_id;
+	src->parent_mat_hypertable_id = parent_mat_hypertable_id;
+	src->htoid = hypertable_oid;
+	src->htpartcolno = hypertable_partition_colno;
+	src->htpartcoltype = hypertable_partition_coltype;
+	src->htpartcol_interval_len = hypertable_partition_col_interval;
+	src->bucket_width = 0;				 /* invalid value */
+	src->bucket_width_type = InvalidOid; /* invalid oid */
+	src->interval = NULL;				 /* not specified by default */
+	src->timezone = NULL;				 /* not specified by default */
+	TIMESTAMP_NOBEGIN(src->origin);		 /* origin is not specified by default */
+}
+
+/*
+ * Check if the supplied OID belongs to a valid bucket function
+ * for continuous aggregates.
+ */
+bool
```

Comment:
```suggestion
static bool
```
Also add it to the beginning of this file and remove from `common.h`